### PR TITLE
Move mob tick code to cChunk

### PR DESCRIPTION
**Pull request by LogicParrot**
After years, the mob tick code moves back to cChunk.

### CODE REVIEW

**Comment by NiLSPACE** [comment]
What are the advantages of having the code in cChunk?

**Comment by LogicParrot** [comment]
- Simplicity. All entities ticks in the chunk, why should mobs be any different?
 - Performance. Every tick, cWorld must hold a lock. Also, it must iterate all entities, which means iterating all chunks for the second time (it has already iterated them for the first time to tick them). Why not take advantage of the already existing chunk tick iteration? 

The real question is: Why is the mob tick code in cWorld? Well, it was put there as a hack to achieve proper mob destruction. This PR can destroy mobs properly.

As for the lock in cWorld::TickMobs, I still need to remove or rewrite MobCensus before removing it.

**Comment by LogicParrot** [comment]
Relevant issues: #277 #283

File: src/Chunk.cpp [hunk 1]
```diff
@@ -1864,13 +1863,41 @@ void cChunk::RemoveClient(cClientHandle * a_Client)
 		}
 	}
 
+	// All clients were removed. No one sees this chunk any more.
+	// Delete all hostiles inside it.
+	if (!HasAnyClients())
+	{
+		for (cEntityList::iterator itr = m_Entities.begin(); itr != m_Entities.end(); ++itr)
```
Comment (tigerw) [inline src/Chunk.cpp]: The C++ standards committee expresses its dissatisfaction at the non-use of C++11 features here.

File: src/Chunk.cpp [hunk 2]
```diff
@@ -1864,13 +1863,41 @@ void cChunk::RemoveClient(cClientHandle * a_Client)
...
+		{
+			cEntity * Entity = *itr;
+			if (Entity->IsTicking() && Entity->IsMob())
+			{
+				auto Monster = static_cast<cMonster *>(Entity);
+				if (
+					(Monster->GetMobFamily() == cMonster::eFamily::mfHostile) &&
+					(Monster->GetCustomName() == "")  // Monsters that have name tags are never removed
```
Comment (tigerw) [inline src/Chunk.cpp]: The writers of `std::string::empty` express their dissatisfaction at the non-use of this function here.

File: src/Chunk.cpp [hunk 3]
```diff
@@ -1864,13 +1863,41 @@ void cChunk::RemoveClient(cClientHandle * a_Client)
...
+					}
+				}
+			}
+		}
+	}
 	return;
 }
 
 
-
```
Comment (tigerw) [inline src/Chunk.cpp]: @madmaxoft expresses his dissatisfaction at your non-compliance with the style guide here.

File: src/Chunk.cpp [hunk 4]
```diff
@@ -1864,13 +1863,41 @@ void cChunk::RemoveClient(cClientHandle * a_Client)
...
+						auto Wolf = static_cast<cWolf *>(Monster);
+						if (Wolf->IsAngry())
+						{
+							Monster->Destroy(true);
+						}
+					}
+				}
+			}
+		}
+	}
```
Comment (tigerw) [inline src/Chunk.cpp]: Braces here, braces there, braces everywhere!

**Comment by LogicParrot** [comment]
@tigerw No more conflicts any time soon. Promise :)

**Comment by LogicParrot** [comment]
Sorry, merged this on accident. I want to make sure all pointers behave right first.

**Pull request merged**